uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-k8s/src/main/java/org/uberfire/java/nio/fs/k8s/K8SFileSystemProvider.java

```diff
@@ -0,0 +1,337 @@
+
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.uberfire.java.nio.IOException;
+import org.uberfire.java.nio.base.AbstractBasicFileAttributeView;
+import org.uberfire.java.nio.base.BasicFileAttributesImpl;
+import org.uberfire.java.nio.base.GeneralPathImpl;
+import org.uberfire.java.nio.channels.SeekableByteChannel;
+import org.uberfire.java.nio.file.AccessDeniedException;
+import org.uberfire.java.nio.file.AccessMode;
+import org.uberfire.java.nio.file.CopyOption;
+import org.uberfire.java.nio.file.DeleteOption;
+import org.uberfire.java.nio.file.FileAlreadyExistsException;
+import org.uberfire.java.nio.file.FileStore;
+import org.uberfire.java.nio.file.LinkOption;
+import org.uberfire.java.nio.file.NoSuchFileException;
+import org.uberfire.java.nio.file.NotDirectoryException;
+import org.uberfire.java.nio.file.OpenOption;
+import org.uberfire.java.nio.file.Path;
+import org.uberfire.java.nio.file.attribute.BasicFileAttributes;
+import org.uberfire.java.nio.file.attribute.FileAttribute;
+import org.uberfire.java.nio.file.attribute.FileAttributeView;
+import org.uberfire.java.nio.fs.cloud.CloudClientFactory;
+import org.uberfire.java.nio.fs.file.SimpleFileSystemProvider;
+
+import static org.kie.soup.commons.validation.PortablePreconditions.checkCondition;
+import static org.kie.soup.commons.validation.PortablePreconditions.checkNotNull;
+import static org.uberfire.java.nio.fs.k8s.K8SFileSystemConstants.CFG_MAP_ANNOTATION_FSOBJ_SIZE_KEY;
+import static org.uberfire.java.nio.fs.k8s.K8SFileSystemConstants.CFG_MAP_FSOBJ_CONTENT_KEY;
+import static org.uberfire.java.nio.fs.k8s.K8SFileSystemConstants.K8S_FS_SCHEME;
+import static org.uberfire.java.nio.fs.k8s.K8SFileSystemUtils.createOrReplaceFSCM;
+import static org.uberfire.java.nio.fs.k8s.K8SFileSystemUtils.createOrReplaceParentDirFSCM;
+import static org.uberfire.java.nio.fs.k8s.K8SFileSystemUtils.deleteAndUpdateParentCM;
+import static org.uberfire.java.nio.fs.k8s.K8SFileSystemUtils.getFsObjCM;
+import static org.uberfire.java.nio.fs.k8s.K8SFileSystemUtils.getPathByFsObjCM;
+import static org.uberfire.java.nio.fs.k8s.K8SFileSystemUtils.isDirectory;
+import static org.uberfire.java.nio.fs.k8s.K8SFileSystemUtils.isRoot;
+
+public class K8SFileSystemProvider extends SimpleFileSystemProvider implements CloudClientFactory {
+    private static final Logger logger = LoggerFactory.getLogger(K8SFileSystemProvider.class);
+    public K8SFileSystemProvider() {
+        super(null, OSType.UNIX_LIKE);
+        this.fileSystem = new K8SFileSystem(this, K8SFileSystem.UNIX_SEPARATOR_STRING);
+    }
+
+    @Override
+    public String getScheme() {
+        return K8S_FS_SCHEME;
+    }
+
+    @Override
+    public InputStream newInputStream(final Path pathIn,
+                                      final OpenOption... options) {
+        checkNotNull("path", pathIn);
```

Comment:
if we check `checkNotNull` javadoc, it takes the name of the parameter as first argument. we need in all this class update all `checkNotNull` invocations to be consistent with the javadoc.

Reply:
changed as per suggested